### PR TITLE
[multicast_dns] Fix for #88242 README.md broken link for bin dir

### PR DESCRIPTION
**Pull request by guyluz11**
Fix for [#88242](https://github.com/flutter/flutter/issues/88242) multicast_dns README.md broken link for bin dir

## Pre-launch Checklist

- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated `CHANGELOG.md` to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the `#hackers-new` channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: packages/multicast_dns/README.md [hunk 1]
```diff
@@ -18,4 +18,4 @@ import 'package:multicast_dns/multicast_dns.dart';
 
 Then use the `MDnsClient` Dart class in your code. To see how this is done,
 check out the [example app](example/main.dart) or the sample implementations in
-the [bin](bin/) directory.
+the [example](example/) directory.
```
Comment (stuartmorgan) [inline packages/multicast_dns/README.md:21]: This is the example app, which was already just mentioned and linked to.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
This needs a version update, as explained in the PR template checklist that you deleted.

Please don't delete that checklist next time; it's there for a reason.

**Comment by guyluz11** [comment]
> This needs a version update, as explained in the PR template checklist that you deleted.
> 
> Please don't delete that checklist next time; it's there for a reason.

Will add it back

**Review by stuartmorgan — APPROVED** [review]
I adjusted the version change to be a bug fix rather than minor feature release. Other than that, LGTM. Thanks for the contribution!

**Pull request merged**